api/sirsidynix_authentication_provider.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import annotations
+
+from gettext import gettext as _
+from typing import TYPE_CHECKING
+from urllib.parse import urljoin
+
+from api.authenticator import BasicAuthenticationProvider, PatronData
+from core.model.configuration import ExternalIntegration
+from core.util.http import HTTP
+
+if TYPE_CHECKING:
+    from requests import Response
+
+
+class SirsiDynixHorizonAuthenticationProvider(BasicAuthenticationProvider):
+    NAME = "SirsiDynix Horizon Authentication"
+    DESCRIPTION = "SirsiDynix Horizon Webservice Authentication"
+    FLOW_TYPE = "http://librarysimplified.org/authtype/sirsidynix-horizon"
+
+    class Keys:
+        APP_ID = "APP_ID"
+        CLIENT_ID = "CLIENT_ID"
+
+    SETTINGS = [
```

Comment:
We should probably provide an appropriate `LIBRARY_SETTINGS` field, so that we can set a Library ID per library associate with this authentication mechanism. 

See the `LIBRARY_SETTINGS` in the base class https://github.com/ThePalaceProject/circulation/blob/main/api/authenticator.py#L1353-L1428 and what we are doing in the `MilleniumPatronAPI` class https://github.com/ThePalaceProject/circulation/blob/main/api/millenium_patron.py#L170-L187